### DISCUSSION TITLE
Allow dbeaver git to use the ssh agent.

### DISCUSSION THREAD

**Discussion by skotos** [dbeaver/dbeaver#11732] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Pulling and Pushing from a git remote is not possible with private key authentication with either JSch or Apache MINA sshd

**Describe the solution you'd like**
Could we have the option of requesting the OS SSH agent to handle the ssh connection for the DBeaver git module? Similar to the option we have for db connections over ssh tunnelling.

**Describe alternatives you've considered**
Git pushes and pulls happen in a terminal outside of the IDE.  (We cannot use password auth on our git server.)

**Additional context**
![dbauth](https://user-images.githubusercontent.com/1259908/102444944-781aa080-407e-11eb-84bd-ed03dfebd060.png)
![gitauth](https://user-images.githubusercontent.com/1259908/102444946-794bcd80-407e-11eb-8642-ba01deb93036.png)

**Comment by kseniiaguzeeva** [comment]
Thank you for the good idea!

**Comment by skotos** [comment]
Thanks!  And thanks for a great project.

**Comment by serge-rider** [comment, marked as answer]
DBeaver uses EGit for Git integration.
Generally it already supports SSH agents: https://stackoverflow.com/questions/13396405/eclipse-git-and-ssh-agent
There are known issues with different agents: https://bugs.eclipse.org/bugs/show_bug.cgi?id=179924

Anyhow we can't fix it from DBeaver side as we rely on EGit implementation.

**Reply by sskras** [reply]
@serge-rider [ on Mar 3, 2021 ](#discussioncomment-482960) 

> DBeaver uses EGit for Git integration.
> Generally it already supports SSH agents: https://stackoverflow.com/questions/13396405/eclipse-git-and-ssh-agent
> There are known issues with different agents: https://bugs.eclipse.org/bugs/show_bug.cgi?id=179924

This ticket points to a newer one https://bugs.eclipse.org/bugs/show_bug.cgi?id=541274, which says:

    [Product:](https://bugs.eclipse.org/bugs/describecomponents.cgi)	JGit

Now quoting some [forum post](https://www.eclipse.org/forums/index.php/t/273443/#msg_775078):

> JGit is the Java implementation of Git. It is a library, that also can be used in your own applications. It also provides some sort of CLI operations.
> EGit on the other side is the Eclipse team provider plugin for Git, which uses JGit as Git implementation. Simplified you could say EGit is the UI part, and JGit the background part.
> JGit doesn't depend on EGit, but EGit does depend on JGit.
> <sub>[Updated on: Thu, 05 January 2012 09:39]</sub>

Could it be that **dbeaver** also uses **JGit** under the hood?

I would file a request (about its' ability to use `ssh-agent` implementations from Cygwin and MSYS2), but I try to be sure about the component.

**Comment by I33Buckler** [comment]
It is already 2024 and I find that this is the most recent comment on a still existing problem. Is there any active workaround as I have no password login for ssh.
